### PR TITLE
Fix Groups Usage

### LINKED ISSUE #9: The "Basic demo" done in React doesn't work.
Hi,

I have created a [very simple React app here](https://github.com/SukantGujar/test-timeline) which is basically a (visjs setup-wise) clone of the [Basic demo](http://visjs.org/examples/timeline/basicUsage.html) available on visjs documentation site.

However for some reason, the items are not displayed in the React app. I compared the DOM and noticed that there are differences in the vis-group and vis-content nodes - Both are empty in the React app while they contain child nodes in the demo page.

The "Basic demo" looks like this - 
image
And the React test app looks like this -
image

Any ideas?

I created this small app as a testbed to isolate rendering issues we faced while integrating react-visjs-timeline component in a larger React app.

Note that both the React app and the Basic demo use the same version of visjs library (4.18.0).

Thanks

### PR DESCRIPTION
**Pull request by willmcclellan**
Fixes issues with passing empty groups. Also tided up code a little.

Fixes #9, #5 

### Updates

* Groups are not applied if not passed
* Don't wrap data in `DataSet` as it's unnecessary. We don't use 2-way binding in react context and vis supports plain JS arrays
* Don't set all options in constructor as it doesn't behave consistently. Modifying options with methods is also cleaner as we reuse the same code on update
* Added `examples` directory which uses examples from visjs to validate usage. Just uses create-react-app which does the trick, but something more lightweight would be preferable

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 build
```
Comment (jonathanchrisp) [inline .gitignore]: Might be worth just using the node .gitignore that Github has [here](https://github.com/github/gitignore/blob/master/Node.gitignore)

File: examples/src/App.test.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
```
Comment (jonathanchrisp) [inline examples/src/App.test.js]: Is it worth including this file?

File: examples/src/logo.svg [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 841.9 595.3">
```
Comment (jonathanchrisp) [inline examples/src/logo.svg]: This doesn't appear to be used but I suppose this doesn't matter.

**Comment by willmcclellan** [comment]
Thanks @jonathanchrisp!

**Pull request merged**